### ISSUE TITLE
Hygiene breaks add_one! macro

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/proc-macro-hack#15]
The `add_one!` example from the readme works with nightly-2017-12-02 but not nightly-2017-12-03.

```
   Compiling example v0.0.0
error[E0425]: cannot find value `x` in this scope
 --> src/main.rs:8:16
  |
8 |     let nine = add_one!(x) + add_one!(2 + 3);
  |                ^^^^^^^^^^^ not found in this scope
  |
```

**Comment by rushmorem** [comment]
Is there a workaround for this?

**Comment by tcr** [comment]
@rushmorem No workaround exists. As a result of https://github.com/rust-lang/rust/issues/46478 it was decided that the fix for macro hygiene will stay, disallowing the use of local variables as macro input to proc-macro-hack.

As of 727987f54abf242e58c685705b4dcd59001fdeb7 add_one() in the README no longer references local variables anymore.

**Comment by dtolnay** [comment]
Fixed in 0.5.0.
